daily-coding/118/cody.py

```diff
@@ -0,0 +1,66 @@
+def squareSort(numbers):
+    i = 0
+    j = len(numbers) - 1
+
+    if len(numbers) == 0:
+        return numbers
+
+    # all numbers are positive
+    if numbers[i] >= 0:
+        for m in range(len(numbers)):
+            numbers[m] = numbers[m]**2
+        return numbers
+
+    # all numbers are negative
+    if numbers[j] < 0:
+        m = j
+        mid = len(numbers)//2
+        for n in range(mid):
+            numbers[m], numbers[n] = numbers[n]**2, numbers[m]**2
+            m -= 1
+        if len(numbers) % 2 != 0:
+            numbers[mid] = numbers[mid]**2
+        return numbers
+
+    # some numbers are negative and some are positive
+    negatives = []
+    positives = []
+    k = 0
+    while numbers[k] < 0:
+        negatives.append(numbers[k]**2)
+        k += 1
+
+    while k < len(numbers):
+        positives.append(numbers[k]**2)
+        k += 1
+
+    r = len(negatives) - 1
+    s = 0
+    t = 0
+    
+    while r >= 0 and s < len(positives):
+        if negatives[r] < positives[s]:
+            numbers[t] = negatives[r]
+            r -= 1
+        else:
+            numbers[t] = positives[s]
+            s += 1
+        t += 1
+
+    while r >= 0:
+        numbers[t] = negatives[r]
+        r -= 1
+        t += 1
+
+    while s > len(positives):
+        numbers[t] = positives[s]
+        s += 1
+        t += 1
+
+    return numbers
+
+print(squareSort([-50, -20, -10]))
```

Comment:
Could you uncomment these tests and add your new ones to my solution?

Reply:
Done.